sys/dev/hptmv/ioctl.c

```diff
@@ -87,7 +87,9 @@ static int
 event_queue_add(PHPT_EVENT pEvent)
 {
 	int p;
-	p = (event_queue_tail + 1) % MAX_EVENTS;
+	p = (event_queue_tail + 1);
+	if (p == MAX_EVENTS)
+		p = 0;
```

Comment:
Modulo doesn't necessarily yield zero, it can introduce some weird bugs, but I'm not completely sure.

Reply:
This one is likely bad. The old code is perfectly fine.